src/Resources/Resource.php

```diff
@@ -23,5 +27,8 @@ protected function fill()
         foreach ($this->attributes as $key => $value) {
             $this->{$key} = $value;
         }
+        if(property_exists($this,'event_id') && $this->event_id) {
+            $this->event = $this->spinupwp->events->get($this->event_id);
+        }
```

Comment:
We don't want to be performing an API call when the class is instantiated. This should be lazy-loaded. Meaning, we only make the API call when the user specifically requests the event on the resource.

`$site->event()`

Reply:
Yeah that's exactly what it is